### ISSUE TITLE
Document how to build on Mac

### ISSUE THREAD

**Issue by EricEdens** [corretto/corretto-8#119]
Similar to https://github.com/corretto/corretto-8/issues/117, let's document how to build on Mac.

**Comment by chrisvire** [comment]
I am trying to build.  I get:

```
configure: error: Xcode 4 or 6 is required to build JDK 8, the version found was 11.5. Use --with-xcode-path to specify the location of Xcode or make Xcode active by using xcode-select.
```

**Comment by chrisvire** [comment]
I also get this error:

```
FAILURE: Build failed with an exception.
> Task :checkJfxrtJar FAILED

* Where:
Build file '/Users/hubbard/codereview/amazon/corretto-8/javafx/corretto-build/buildRoot/rt-8u202-ga/build.gradle' line: 221

* What went wrong:
Execution failed for task ':checkJfxrtJar'.
> FAIL: /Library/Java/JavaVirtualMachines/amazon-corretto-8.jdk/Contents/Home/jre/lib/ext/jfxrt.jar must be removed before building sdk
```
Really?

**Comment by chrisvire** [comment]
@ziyiluo, @EricEdens

I have tried (which took a significant amount of time):
* Setup Mac OS X Yosemite VM in Parallels
* Installed Xcode 6.4
* ran `./gradew installers:mac:pkg:build`

I got an error about freetype missing:
```
configure: error: Could not find freetype!  
configure exiting with result code 1
```
I tried to install homebrew and use that to install freetype. I had to run `xcode-select` to reset to `/Applications/Xcode.app`. I stell get the error about freetype.

Do you have a CI/CD configuration (docker image, ansible script, or ???) to have the correct environment to build?

I am trying to test a patch (https://hg.openjdk.java.net/jdk/jdk15/rev/9cfa0137612f) for:
https://bugs.openjdk.java.net/browse/JDK-8244818 
[macos] Java2D Queue Flusher crash while moving application window to external monitor

**Comment by chrisvire** [comment]
@ziyiluo:  I saw https://github.com/corretto/corretto-8/issues/34#issuecomment-453674834:

I tried installing Xcode 4.6 on Yosemite and was stopped with this error:

```
Undefined symbols for architecture x86_64:
  "__Unwind_Resume", referenced from:
      _main in main-ILbFGt.o
  "start", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by chrisvire** [comment]
I tried installing Xcode 5 and got this:
```
configure: error: Xcode 4 or 6 is required to build JDK 8, the version found was 5.0. Use --with-xcode-path to specify the location of Xcode or make Xcode active by using xcode-select.```

And this error:
```
> Task :graphics:linkMacPrismSW FAILED
Undefined symbols for architecture x86_64:
  "___sincos_stret", referenced from:
      _piscesmath_btan in PiscesMath.obj
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by ziyiluo** [comment]
> I also get this error:
> 
> ```
> FAILURE: Build failed with an exception.
> > Task :checkJfxrtJar FAILED
> 
> * Where:
> Build file '/Users/hubbard/codereview/amazon/corretto-8/javafx/corretto-build/buildRoot/rt-8u202-ga/build.gradle' line: 221
> 
> * What went wrong:
> Execution failed for task ':checkJfxrtJar'.
> > FAIL: /Library/Java/JavaVirtualMachines/amazon-corretto-8.jdk/Contents/Home/jre/lib/ext/jfxrt.jar must be removed before building sdk
> ```
> 
> Really?

Yes! In order to build OpenJDK8, you need to create a customized bootstrap JDK **without** `jre/lib/ext/jfxrt.jar`.

**Comment by chrisvire** [comment]
@ziyiluo Would it be possible to work through this in a chat or video call?

**Comment by ziyiluo** [comment]
> I tried installing Xcode 5 and got this:
> 
> ```
> configure: error: Xcode 4 or 6 is required to build JDK 8, the version found was 5.0. Use --with-xcode-path to specify the location of Xcode or make Xcode active by using xcode-select.
> ```
> 
> And this error:
> 
> ```
> > Task :graphics:linkMacPrismSW FAILED
> Undefined symbols for architecture x86_64:
>   "___sincos_stret", referenced from:
>       _piscesmath_btan in PiscesMath.obj
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

Here's a quick guide for building Corretto-8 on macOS. Internally Corretto 8 is built inside a VM instance for reproducibility and security reasons.

Base Image: OSX Yosemite
Xcode: 6.4 (We used 4.x before and then updated to 6.4 to meet the minimum SDK requirement for notarization)

Build the following libraries (latest version) from source (`./configure && make && sudo make install`):
* [autoconf](https://www.gnu.org/software/autoconf/#downloading)
* [freetype](https://www.freetype.org/download.html)

Download the [latest Amazon Corretto 8](https://corretto.aws/downloads/latest/amazon-corretto-8-x64-macos-jdk.pkg) as bootstrap JDK and remove `jre/lib/ext/jfxrt.jar`.

Try to build Corretto-8 now :)
```bash
# Corretto 8 is supported on 10.10 and later. Failed to export MACOSX_DEPLOYMENT_TARGET might be the reason of the linking error you observed.
export MACOSX_DEPLOYMENT_TARGET=10.10
export EXTRA_CORRETTO_CONFIG_OPTIONS="--with-freetype-include=/usr/local/include/freetype2 --with-freetype-lib=/usr/local/lib"
# Due to a recent commit, the extra configure flags are passed as a property now. Previously they are read from the global environment.
./gradlew -Pcorretto.extra_config="$EXTRA_CORRETTO_CONFIG_OPTIONS" ":installers:mac:tar:packaging"
```

**Comment by ziyiluo** [comment]
> @ziyiluo Would it be possible to work through this in a chat or video call?

Hi, Github is our preferred channel for communicating non-security-related issues. It makes other developers easier to find answers when they have the same question in the future. I'm available for answering your questions during the work hours in Pacific Daylight Time (PDT). I'll keep an eye on this issue to assist you with any follow-up questions.

**Comment by chrisvire** [comment]
@ziyiluo Thank you so much! I was able to build and will now test the issue that I am having.

**Comment by ziyiluo** [comment]
Glad to hear :)
